equinix/resource_network_device.go

```diff
@@ -1538,6 +1548,11 @@ func getNetworkDeviceStateChangeConfigs(c ne.Client, deviceID string, timeout ti
 			createNetworkDeviceStatusResourceUpgradeWaitConfiguration(c.GetDevice, deviceID, 5*time.Second, timeout),
 		)
 	}
+	if _, found := changes[neDeviceClusterSchemaNames["ClusterName"]]; found {
```

Comment:
There won't be any device status changes when changing cluster name, and thus we can remove the status check here.